utilities/PerformanceTesting/query.py

```diff
@@ -0,0 +1,289 @@
+        resp = requests.get(url, headers=headers, params=params, verify=False)
+
+        resp.raise_for_status()
+
+        return resp.json()
+
+    def query_and_get_value(self, query):
+        """
+        Perform a Prometheus query and return the value from it
+        in the following format {value: 'value'}
+
+        :param query: A Prometheus query
+
+        :return: The value returned by the Prometheus query
+        in the following format {value: 'value'}
+        """
+        response = self.query(query)
+
+        try:
+            # Example response:
+            # {
+            #   "status": "success",
+            #   "data": {
+            #     "resultType": "vector",
+            #     "result": [
+            #       {
+            #         "metric": {
+            #           "job": "kubelet"
+            #         },
+            #         "value": [
+            #           1650322259.04,
+            #           "259264.2228070175"
+            #         ]
+            #       }
+            #     ]
+            #   }
+            # }
+
+            return float(response['data']['result'][0]['value'][1])
+        except (KeyError, ValueError, IndexError):
+            return None
+
+    def get_stats_for_query(self, query):
+        """
+        Gets statics in g_stats_names for a query, by adding {stat} by (job)
+        to the begining of the query.
+
+        :param query: The query for which the statics are obtained
+
+        :return: A json with statics
+        """
+        result = {}
+        for stat in g_stat_names:
+            stat_query = f'{stat} by (job) {query}'
+            result[g_stat_names[stat]] = self.query_and_get_value(stat_query)
+
+        return result
+
+
+def get_collector_timers(querier):
+    """
+    Performs Prometheus queries relevant to Collector timers
+
+    :param querier: An object containing information
+    needed to query the Prometheus API
+
+    :return: Json containing information about Collector timers
+    """
+    res = {}
+    timers = [
+                "net_scrape_update",
+                "net_scrape_read",
+                "net_write_message",
+                "net_create_message"
+            ]
+
+    total_time = {}
+    for stat in g_stat_names:
+        total_time[g_stat_names[stat]] = 0
+
+    for timer in timers:
+        res[timer] = {'description': f'Time taken by {timer}', 'units': 'microseconds'}
+        for stat in g_stat_names:
+            time_query = f'{stat} by (job) (rox_collector_timers{{type="{timer}_times_us_avg"}})'
+
+            time_value = querier.query_and_get_value(time_query)
+            res[timer][g_stat_names[stat]] = time_value
+
+            try:
+                total_time[g_stat_names[stat]] += time_value
+            except TypeError:
+                err_msg = "WARNING: The following query did not return a valid response\n"
+                sys.stderr.write(err_msg)
+                sys.stderr.write(time_query)
```

Comment:
Don't we need to add a new line to this?
```suggestion
                sys.stderr.write(f'{time_query}\n')
```

Reply:
Done